Packages/com.unity.inputsystem/InputSystem/Controls/IOptimizedControl.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace UnityEngine.InputSystem.Controls
+{
+    public enum OptimizedControlDataType
```

Comment:
Not sure about this design. Users wouldn't be able to optimize their own controls with this. Can it not be an actual Type type?

Reply:
I want to easily switch case on the value, so enum would be better/faster.
It doesn't have to be public, we can make it internal.

Reply:
Not sure about users optimizing their own controls, this feature would be internal for a time being, if we get nudged we might consider opening it.

In general I'm not convinced that user defined types are a good idea, that greatly limits what we can do with regards to editor UX and data structures. If all controls types are known at our compile time we would be able to do much more effective architecture.

Users defining their own devices is different and should stay as-is.

Reply:
tried with Type and the checks are much more expensive it seems https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcDMACR2DC2A3utmbjgKYB2ArlNgCqUTACidUp5Ja5/2AILYAvNgAMAGm4DsAIWl9ZBGdgC+AblWqADgCcAlgDcwwSkxbtO2M6w71RFu5wB0grUrL7jp84wCeOua2wAFBjsCBlAD2AGYAFAbUwACUHvyqWNgARtHRADbY9lD4ABaUAMYA1vEpxKr8cADsNpbFomLMzvQu+OnkauiZOLkFTFFllTV1vMq4LSFh5iJikUFx8bH50aZpqoNoakA===

Reply:
After some discussion, we can switch to int (FourCC) the same way as `InputStateBlock` works

Reply:
Fixed